mypy/util.py

```diff
@@ -255,3 +257,49 @@ def hard_exit(status: int = 0) -> None:
     sys.stdout.flush()
     sys.stderr.flush()
     os._exit(status)
+
+
+# The following is a backport of stream redirect utilities from Lib/contextlib.py
```

Comment:
Please also add a comment explaining this can be deleted once we stop supporting Python 3.4.